### PR TITLE
FPGA version: DIE type and speed grade

### PR DESCRIPTION
**Pull request by t-salminen**
FPGA DIE type and speed grade information is available in fgpa version register area:
https://github.com/tiiuae/saluki-fpga/tree/main/ssrc_saluki-pi#fpga-version-info-register

This information is appended to FPGA version string. E.g. `FPGA: 2.65535 (0xffff0243), MPFS250 (SG 1)`

Also `component_version1`-field size increased to match `static char fpga_info[64]` in `platforms/nuttx/src/px4/microchip/mpfs/version/board_mcu_version.c`

```
saluki> ver all
HW arch: SSRC_SALUKI_PI
HW type: Saluki003000
HW version: 0x003
HW revision: 0x000
PX4 git-hash: df811b9bd62aac1eba34f9b3d2c643529724bc79
PX4 version: 1.14.0 0 (17694720)
PX4 git-branch: mpfs_die_type
Bootloader version: 1.4.0
Accelerator version: FPGA: 2.65535 (0xffff0243), MPFS250 (SG 1)
Vendor version: 12.0.1 0 (201326848)
OS: NuttX
OS version: Release 12.2.1 (201458175)
OS git-hash: d493bcdf064c25b0d145a5bd88f3a2af4cd8dd9f
Build datetime: May 29 2024 08:35:00
Build uri: localhost
Build variant: flat
Toolchain: GNU GCC, 10.2.0
PX4GUID: 20003de8b10fd67016d5b11afecd077b7147
MCU: (null), rev.  
saluki> 
```

### CODE REVIEW

**Comment by t-salminen** [comment]
@kordets , you tried to test this some time ago. With this change the test should be possible.

File: msg/SystemVersionString.msg [hunk 1]
```diff
@@ -5,5 +5,5 @@ char[32] hw_version		# main autopilot hw version
 char[32] sw_version		# main autopilot sw version
 char[32] os_version		# OS or middleware version
 char[32] bl_version		# bootloader version
-char[32] component_version1	# vendor specific component version
```
Comment (jlaitine) [inline msg/SystemVersionString.msg:8]: This looks sensible, but I wouldn't do this still. I believe it requires re-compiling the linux-side uXRCE-DDS type definitions for the mission computer to match the definition. Maybe we could reduce the size of the hw_info and fpga_info and print out the accelerator(FPGA) version in a bit shorter form (e.g. remove the FPGA: string) to fit into 32 bytes?

**Review by jlaitine — COMMENTED** [review]
One minor comment; I'd like to keep the type string shorter. Otherwise LGTM

File: msg/SystemVersionString.msg (hunk 1 above)
Comment (t-salminen) [reply msg/SystemVersionString.msg]: Right. I'll leave this as it was.

**Comment by t-salminen** [comment]
Version output with shrink accelerator version string

```
saluki> ver all
HW arch: SSRC_SALUKI_PI
HW type: Saluki008000
HW version: 0x008
HW revision: 0x000
PX4 git-hash: df811b9bd62aac1eba34f9b3d2c643529724bc79
PX4 version: 1.14.0 0 (17694720)
PX4 git-branch: mpfs_die_type
Bootloader version: 1.5.0
Accelerator version: 4.65535 (0xffff0408), MPFS250 (SG 1)
Vendor version: 12.0.1 0 (201326848)
OS: NuttX
OS version: Release 12.2.1 (201458175)
OS git-hash: d493bcdf064c25b0d145a5bd88f3a2af4cd8dd9f
Build datetime: Jun  4 2024 06:01:32
Build uri: localhost
Build variant: default
Toolchain: GNU GCC, 10.2.0
PX4GUID: 200033fd50480d7ce1f842a0908f2441e17f
MCU: (null), rev.  
saluki> 
```

**Review by jlaitine — APPROVED** [review]

**Pull request merged**